app/src/main/java/eu/kanade/presentation/updates/UpdateUpcomingScreen.kt

```diff
@@ -0,0 +1,304 @@
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.outlined.HelpOutline
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.platform.LocalUriHandler
+import androidx.compose.ui.unit.dp
+import cafe.adriel.voyager.navigator.LocalNavigator
+import cafe.adriel.voyager.navigator.currentOrThrow
+import eu.kanade.presentation.components.AppBarTitle
+import eu.kanade.presentation.components.UpIcon
+import eu.kanade.presentation.components.relativeDateText
+import eu.kanade.presentation.updates.components.calendar.Calendar
+import eu.kanade.tachiyomi.ui.updates.UpdateUpcomingScreenModel
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.ImmutableMap
+import kotlinx.collections.immutable.persistentMapOf
+import kotlinx.coroutines.launch
+import tachiyomi.domain.manga.model.Manga
+import tachiyomi.i18n.MR
+import tachiyomi.presentation.core.components.FastScrollLazyColumn
+import tachiyomi.presentation.core.components.ListGroupHeader
+import tachiyomi.presentation.core.components.TwoPanelBox
+import tachiyomi.presentation.core.components.material.Scaffold
+import tachiyomi.presentation.core.i18n.stringResource
+import java.time.LocalDate
+
+@Composable
+fun UpdateUpcomingScreen(
+    state: UpdateUpcomingScreenModel.State,
+    modifier: Modifier = Modifier,
+    isTabletUi: Boolean = false,
+    onClickUpcoming: (manga: Manga) -> Unit = {},
+) {
+    if (!isTabletUi) {
+        UpdateUpcomingScreenSmallImpl(
+            state = state,
+            modifier = modifier,
+            onClickUpcoming = onClickUpcoming,
+        )
+    } else {
+        UpdateUpcomingScreenLargeImpl(
+            state = state,
+            isTabletUi = isTabletUi,
+            modifier = modifier,
+            onClickUpcoming = onClickUpcoming,
+        )
+    }
+}
+
+@Composable
+internal fun UpdateUpcomingScreenSmallImpl(
+    state: UpdateUpcomingScreenModel.State,
+    modifier: Modifier = Modifier,
+    onClickUpcoming: (manga: Manga) -> Unit = {},
+) {
+    Scaffold(
+        modifier = modifier,
+        topBar = {
+            UpdateUpcomingToolbar()
+        },
+    ) { paddingValues ->
+
+        UpdateUpcomingSmallContent(
+            upcoming = state.items,
+            events = state.events,
+            contentPadding = paddingValues,
+            onClickUpcoming = onClickUpcoming,
+        )
+    }
+}
+
+const val HelpUrl = "https://mihon.app/docs/faq/upcoming"
```

Comment:
we don't have this

Reply:
Was waiting for this to be closer to going in to write the copy per the original description. See https://github.com/mihonapp/website/pull/47 for that page.